crates/rune/src/runtime/value.rs

```diff
@@ -852,10 +852,14 @@ impl Value {
             }
             _ => {
                 let result =
-                    vm_try!(caller.call_protocol_fn(Protocol::STRING_DEBUG, self.clone(), (f,)));
+                    caller.call_protocol_fn(Protocol::STRING_DEBUG, self.clone(), (&mut *f,)); // reborrow f to avoid moving it
 
-                vm_try!(<()>::from_value(result));
-                return VmResult::Ok(());
+                if let VmResult::Ok(result) = result {
+                    vm_try!(<()>::from_value(result));
+                } else {
+                    let type_info = vm_try!(self.type_info());
+                    vm_write!(f, "<{} object at {:p}>", type_info, self);
```

Comment:
Not a bad approach in general, but I am weary of potentially leaking real address information. In itself it shouldn't be usable for anything nefarious, but might maybe be used to defeat e.g. [address space randomization](https://en.wikipedia.org/wiki/Address_space_layout_randomization). Maybe we should stick to type info?

Reply:
I wanted something to make instances a bit more unique, but also the `self` really isn't interesting here since the inner value is a arc-like thing.

Do we have any other info to separate something like `dbg!(foo_a, foo_b);`?

Reply:
Ah, so `self` points to the `Value` struct, and by extension wherever that is stored (the program stack, the vm stack, ...). What you probably meant to do would be to print the address of the pointer stored in [`Shared<ValueKind>`](https://github.com/rune-rs/rune/blob/fc2bc0c6b48893b819dc1681ab08a19655ce77b1/crates/rune/src/runtime/shared.rs#L21)? At least that address should be harder to use to locate interesting things in memory.

If that's also sensitive we could also opt to allocate pseudo-random identifiers (through e.g. a static AtomicUsize) and store them in `SharedBox`. Either enabled through a feature or a debug build.

Reply:
Switched it to use the pointer from self.inner, as a start. I can't claim to care about the security implications  of it still, so I'll defer to you if you'd prefer some other approach. The downside of using a feature/debug/... is that it doesn't have a fallback. I guess we could opt-out of printing addresses with a feature though, or not-in-release.